src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificateAssetDownloader.cs

```diff
@@ -130,55 +130,118 @@ internal static class CertificateAssetDownloader
                 MethodInfo? sendMethod = httpClientType.GetMethod("Send", new Type[] { httpRequestMessageType, typeof(CancellationToken) });
                 PropertyInfo? responseContentProp = httpResponseMessageType.GetProperty("Content");
+                PropertyInfo? responseStatusCodeProp = httpResponseMessageType.GetProperty("StatusCode");
+                PropertyInfo? responseHeadersProp = httpResponseMessageType.GetProperty("Headers");
+                PropertyInfo? responseHeadersLocationProp = httpResponseHeadersType.GetProperty("Location");
                 MethodInfo? readAsStreamMethod = httpContentType.GetMethod("ReadAsStream", Type.EmptyTypes);
-                if (pooledConnectionIdleTimeoutProp == null || requestUriProp == null || httpRequestMessageCtor == null || sendMethod == null || responseContentProp == null || readAsStreamMethod == null)
+
+                if (socketsHttpHandlerCtor == null || pooledConnectionIdleTimeoutProp == null || allowAutoRedirectProp == null || httpClientCtor == null ||
+                    requestUriProp == null || httpRequestMessageCtor == null || sendMethod == null || responseContentProp == null || responseStatusCodeProp == null ||
+                    responseHeadersProp == null || responseHeadersLocationProp == null || readAsStreamMethod == null)
                 {
                     Debug.Fail("Unable to load required member.");
                     return null;
                 }
 
                 // Only keep idle connections around briefly, as a compromise between resource leakage and port exhaustion.
                 const int PooledConnectionIdleTimeoutSeconds = 15;
+                const int MaxRedirections = 10;
 
                 // Equivalent of:
-                // var socketsHttpHandler = new SocketsHttpHandler() { PooledConnectionIdleTimeout = TimeSpan.FromSeconds(PooledConnectionIdleTimeoutSeconds) };
+                // var socketsHttpHandler = new SocketsHttpHandler() {
+                //     PooledConnectionIdleTimeout = TimeSpan.FromSeconds(PooledConnectionIdleTimeoutSeconds),
+                //     AllowAutoRedirect = false
+                // };
                 // var httpClient = new HttpClient(socketsHttpHandler);
-                object? socketsHttpHandler = Activator.CreateInstance(socketsHttpHandlerType);
+                // Note: using a ConstructorInfo instead of Activator.CreateInstance, so the ILLinker can see the usage through the lambda method.
+                object? socketsHttpHandler = socketsHttpHandlerCtor.Invoke(null);
                 pooledConnectionIdleTimeoutProp.SetValue(socketsHttpHandler, TimeSpan.FromSeconds(PooledConnectionIdleTimeoutSeconds));
-                object? httpClient = Activator.CreateInstance(httpClientType, new object?[] { socketsHttpHandler });
+                allowAutoRedirectProp.SetValue(socketsHttpHandler, false);
```

Comment:
I think it'd be worth including a todo comment citing that we should clean up this method if https://github.com/dotnet/runtime/issues/45364 is ever implemented.